acceptance/tests/reports/corrective_change_new_resource.rb

```diff
@@ -0,0 +1,83 @@
+require 'yaml'
+
+test_name "C98092: corrective change new resource" do
+require 'puppet/acceptance/environment_utils'
+extend Puppet::Acceptance::EnvironmentUtils
+
+  app_type = File.basename(__FILE__, '.*')
+  tmp_environment   = mk_tmp_environment_with_teardown(master, app_type)
+
+  tmp_file = {}
+  agents.each do |agent|
+    tmp_file[agent.hostname] = agent.tmpfile(tmp_environment)
+  end
+
+  teardown do
+    step 'clean out produced resources' do
+      agents.each do |agent|
+        on(agent, "rm #{tmp_file[agent.hostname]}", :accept_all_exit_codes => true)
+      end
+    end
+  end
+
+  file_contents     = 'this is a test'
+  step 'create site.pp with utf8 chars' do
+    manifest = <<MANIFEST
+file { '#{environmentpath}/#{tmp_environment}/manifests/site.pp':
+  ensure => file,
+  content => '
+\$test_path = \$::fqdn ? #{tmp_file}
+file { \$test_path:
+  content => @(UTF8)
+    #{file_contents}
+    | UTF8 
+}
+  ',
+}
+MANIFEST
+    apply_manifest_on(master, manifest, :catch_failures => true)
+  end
+
+  step 'run agent(s)' do
+    with_puppet_running_on(master, {}) do
+      agents.each do |agent|
+        fqdn = agent.hostname
+        config_version = ''
+        config_version_matcher = /configuration version '(\d+)'/
+        on(agent, puppet("agent -t --environment #{tmp_environment} --server #{master.hostname}"),
+           :acceptable_exit_codes => 2).stdout do |agent_out|
+          config_version = agent_out.match(config_version_matcher)
+        end
+        on(agent, "cat #{tmp_file[fqdn]}").stdout do |result|
+          assert_equal(file_contents, result, 'file contents did not match accepted')
+        end
+        on(agent, "rm #{tmp_file[fqdn]}", :accept_all_exit_codes => true)
+        on(agent, puppet("agent -t --environment #{tmp_environment} --server #{master.hostname} --use_cached_catalog"),
+           :acceptable_exit_codes => 2).stdout do |agent_out|
+          assert_match(config_version_matcher, result, 'agent did not use cached catalog')
+        end
+        on(agent, "cat #{tmp_file[fqdn]}").stdout do |result|
+          assert_equal(file_contents, result, 'file contents did not match accepted')
+        end
+      end
+    end
+  end
+
+  # Open last_run_report.yaml
+  step 'Check report' do
+     agents.each do |agent|
+       on(agent, puppet('config print statedir')) do |command_result|
+         report_path = command_result.stdout.chomp + '/last_run_report.yaml'
+         on(agent, "cat #{report_path}") do |report_yaml|
+           report_yaml = YAML.load(report_yaml.stdout.gsub(/\!ruby[^\n]*/, ''))
```

Comment:
The gsub statement might be confusing to someone else reading this code for the first time, it would be worth adding a code comment here to explain that we're stripping out ruby class definitions from the report yaml